client/js/views/visualize.react.jsx

```diff
@@ -35,16 +35,19 @@ var Visualize = React.createClass({
     $.getJSON('repos/'+repoFullName+'/commits', {accessToken: window.localStorage.gitHubAccessToken})
     .success(function(commits) {
       if (commits.msg === 'auth required') return window.location = commits.authUrl;
-      if (!Array.isArray(commits)) this.transitionTo('/'); //TODO show error msg first
```

Comment:
if (!Array.isArray(commits)) return this.transitionTo('/', null, {error: 'badRepo'}); //TODO show error msg first